solidity/contracts/ECDSABackportRewards.sol

```diff
@@ -0,0 +1,115 @@
+
+                           Trust math, not hardware.
+*/
+
+pragma solidity 0.5.17;
+
+import "@keep-network/keep-core/contracts/Rewards.sol";
+import "./BondedECDSAKeepFactory.sol";
+import "./api/IBondedECDSAKeep.sol";
+
+contract ECDSABackportRewards is Rewards {
+    // BondedECDSAKeepFactory deployment date, May-13-2020 interval started.
+    // https://etherscan.io/address/0x18758f16988E61Cd4B61E6B930694BD9fB07C22F
+    uint256 internal constant bondedECDSAKeepFactoryDeployment = 1589408351;
+
+    // We are going to have one interval, with a weight of 100%.
+    uint256[] internal backportECDSAIntervalWeight = [100];
+
+    // Interval is the difference in time of creation between older and newer
+    // versions of BondedECDSAKeepFactory.
+    // Older: https://etherscan.io/address/0x18758f16988E61Cd4B61E6B930694BD9fB07C22F
+    // Newer: https://etherscan.io/address/0xA7d9E842EFB252389d613dA88EDa3731512e40bD
+    uint256 internal constant backportECDSATermLength = 123 days; // 10678946 sec
+
+    uint256 internal constant minimumECDSAKeepsPerInterval = 40;
+
+    BondedECDSAKeepFactory factory;
+
+    constructor(address _token, address payable _factoryAddress)
+        public
+        Rewards(
+            _token,
+            bondedECDSAKeepFactoryDeployment,
+            backportECDSAIntervalWeight,
+            backportECDSATermLength,
+            minimumECDSAKeepsPerInterval
+        )
+    {
+        factory = BondedECDSAKeepFactory(_factoryAddress);
+    }
+
+    function _getKeepCount() internal view returns (uint256) {
+        return factory.getKeepCount();
+    }
+
+    function _getKeepAtIndex(uint256 index) internal view returns (bytes32) {
+        return fromAddress(factory.getKeepAtIndex(index));
+    }
+
+    function _getCreationTime(bytes32) internal view returns (uint256) {
+        // Assign each keep to the starting timestamp of its interval.
+        return startOf(0);
+    }
+
+    function _isClosed(bytes32) internal view returns (bool) {
+        // Even though we still have some of the keeps opened, all the keeps
+        // created between May 13 2020 - Sep 14 2020 are considered closed. Because
+        // of the deposits pause https://tbtc.network/news/2020-05-21-details-of-the-tbtc-deposit-pause-on-may-18-2020/
+        // closing all the keeps is not easily achievable. However, we do not want
+        // to block rewards distribution for good stakers caused by the incident on May 18th.
+        return true;
+    }
+
+    function _isTerminated(bytes32 groupIndexBytes)
+        internal
+        view
+        returns (bool)
+    {
+        return false;
+    }
+
+    // A keep is recognized if it was created by this factory.
+    function _recognizedByFactory(bytes32 _keep) internal view returns (bool) {
+        return factory.getKeepOpenedTimestamp(toAddress(_keep)) != 0;
+    }
+
+    function _distributeReward(bytes32 _keep, uint256 amount)
+        internal
+        isAddress(_keep)
```

Comment:
Do we need this check? This function is called from `Rewards` abstract contract only when `_recognizedByFactory` returns `true` (see `factoryMustRecognize` modifier).

If `isAddress` is not needed, then `validAddressBytes` is also not needed.

Reply:
I see your point that by the time we call `_distributeReward`, `_keep` will be already verified, however we're calling this from the abstract contract. If we are not worried that potentially in the future during the upgrades we won't miss that check, then I can remove it.

Reply:
I'd remove it, it's only increasing gas consumption and this contract needs to trust the parent contract.